File: Ahorn/entities/CardShark.jl
module CardShark

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/CardShark" CardShark(x::Integer, y::Integer, cardsPerThrow::Integer=3, detectionRange::Integer=180, health::Integer=2, patrolDistance::Integer=80, throwInterval::Number=1.5)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(CardShark)
)

function Ahorn.selection(entity::CardShark)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CardShark, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/cardShark.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local cardShark = {}

cardShark.name = "MaggyHelper/CardShark"
cardShark.depth = -100
cardShark.fieldInformation = {
    health = {
        minimumValue = 1
    },
    detectionRange = {
        minimumValue = 0
    },
    throwInterval = {
        minimumValue = 0
    },
    cardsPerThrow = {
        minimumValue = 1
    },
    patrolDistance = {
        minimumValue = 0
    }
}
cardShark.placements = {
    {
        name = "normal",
        data = {
            health = 2,
            detectionRange = 180,
            throwInterval = 1.5,
            cardsPerThrow = 3,
            patrolDistance = 80
        }
    }
}

function cardShark.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/card_shark/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function cardShark.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 28, 32, 28)
end

return cardShark
